.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,19 @@
+name: Node.js CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: pnpm/action-setup@v2.0.1 # Using pnpm action to setup pnpm
+        with:
+          version: latest # You can specify a version of pnpm if needed
+      - run: pnpm install --frozen-lockfile # Using pnpm instead of npm
+      - run: pnpm run lint
```

Comment:
we're using `yarn` in all other repos, so sadly I think we should stick to that as our package manager for consistency.